urllib3/contrib/gaepool.py

```diff
@@ -0,0 +1,101 @@
+import warnings
+
+from ..connectionpool import (
+    ConnectionPool,
+    _Default,
+    get_host)
+
+from ..exceptions import (
+    HTTPWarning,
+    TimeoutError,
+)
+
+from ..packages.six import BytesIO
+from ..request import RequestMethods
+from ..response import HTTPResponse
+from ..util.timeout import Timeout
+from ..util.url import Url
+
+try:
+    from google.appengine.api import urlfetch
+except ImportError:
+    urlfetch = None
+
+
+class GAEPlatformWarning(HTTPWarning):
+    pass
+
+
+class GAEConnectionPool(ConnectionPool, RequestMethods):
+
+    def __init__(self, host, port=None, scheme='http', timeout=5, headers=None,
+                 retries=None, validate_certificate=False):
```

Comment:
> Do we want to default to False

Probably not, as the default will be True soon in URLFetch itself.

> Should we emit a warning

No, it's not an error for URLFetch to be provided those arguments, and if we default to True then all http requests will fail.